goodpeer/goodpeer.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+from collections import OrderedDict
+from pyln.client import Plugin, Millisatoshi
+
+plugin = Plugin()
+
+
+def get_channel_infos(chan):
+    """
+    Filter out what we need from a channel entry in listchannels
+    """
+    return {
+        "scid": chan["short_channel_id"],
+        "ppm_fee": chan["fee_per_millionth"],
+        "base_fee": chan["base_fee_millisatoshi"],
+        "value": chan["amount_msat"],
+    }
+
+
+def get_nodes(plugin):
+    """
+    Get public Lightning network nodes.
+
+    :return: A dict with the node id as key, and a list of channels as values.
+    """
+    peers = {}
+    channels = plugin.rpc.listchannels()
+    for chan in channels["channels"]:
+        if chan["fee_per_millionth"] == chan["base_fee_millisatoshi"] == 0:
+            # They are likely an agreement or the same person, they'd just
+            # distort the medians.
+            continue
+        if not chan["active"]:
+            # It could just be temporary, but we don't want zombies!
+            continue
+        id = chan["source"]
+        if id not in peers.keys():
+            peers[id] = [get_channel_infos(chan)]
+        else:
+            peers[id].append(get_channel_infos(chan))
+    return peers
+
+
+def filter_nodes(nodes):
+    """
+    Filter out nodes with <=15 non-dust channels.
+
+    20$ minimum for a bitcoin at 8000$ (250000000 msats).
+    """
+    filtered_nodes = {}
+    dust_limit = Millisatoshi(250000000)
+    for id, channs in nodes.items():
+        if len(list(filter(lambda c: c["value"] > dust_limit, channs))) > 15:
+            filtered_nodes[id] = channs
+    return filtered_nodes
+
+
+def get_medians(channels):
+    """
+    Return the (median_base_fee, median_fees_ppm) of the given peer.
+    """
+    # Loop only once..
+    base_fees = ppm_fees = []
+    for chann in channels:
+        base_fees.append(chann["base_fee"])
+        ppm_fees.append(chann["ppm_fee"])
+    base_fees, ppm_fees = sorted(base_fees), sorted(ppm_fees)
+    n_channs = len(channels)
+    if n_channs % 2 == 0:
+        return base_fees[n_channs // 2], ppm_fees[n_channs // 2]
+    else:
+        median_base = (base_fees[n_channs // 2] +
+                       base_fees[n_channs // 2 + 1]) / 2
+        median_ppm = (ppm_fees[n_channs // 2] +
+                      ppm_fees[n_channs // 2 + 1]) / 2
+        return median_base, median_ppm
+
+
+def reverse_sort(nodes):
+    """
+    Reversely sort the nodes, the lower the score the better.
+    """
+    # Python3.5 hasn't native dicts ordered.. So use OrderedDict
+    sorted_nodes = OrderedDict()
+    for k, v in sorted(nodes.items(),
+                       key=lambda item: item[1]["score"], reverse=True):
+        sorted_nodes[k] = v
+    return sorted_nodes
```

Comment:
Could be simplified to

```python
OrderedDict(sorted(nodes.items(), key=lambda item: -item[1]["score"]))
```

Though it's a bit less descriptive.

Reply:
I prefer it your way, why did you add the minus sign before the score though ?

Reply:
Fixed, I just kept the reverse=True instead of the minus :-)